### PR TITLE
DOCS-#3650: Add optional onboarding instructions and s/onboarding/contributing.

### LINKED ISSUE #3650: Add optional development environment steps to onboarding instructions.
I found the following to be helpful:

- A pre-commit hook requiring flake8 to pass before making any commit.
- Making VSCode format my code with black on save.

I can get a Python Debugger to work in VSCode for personal debug scripts, but not for pytest unit tests, so I won't add instructions on that for now.

### PR DESCRIPTION
**Pull request by mvashishtha**
Signed-off-by: mvashishtha <mahesh@ponder.io>

## What do these changes do?

- Move "onboarding" to "contributing".

- Add instructions to:
  - A pre-commit hook requiring flake8 to pass before making any commit.
  - Make VSCode format code with black on save.

- Reference the new "contributing" doc in the readthedocs.

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #3650  
- [x] tests added and passing
- [x] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

File: onboarding/onboarding.md [hunk 1]
```diff
@@ -76,3 +76,16 @@ you should reopen your terminal to find "(base)" next to your prompt: ![](conda_
...
 
 1. [Add a GPG key](https://docs.github.com/en/authentication/managing-commit-signature-verification/adding-a-new-gpg-key-to-your-github-account ) to your Modin account. Your commits need to be signed with a GPG key. For mac, you can use [Mac GPG](https://gpgtools.org/).
+
+
+1. (Optional) We recommend a few workflow settings:
+
+    1. If you use Visual Studio Code, auto-format with black every time you save changes:
+        1. Install [Microsoft's Python extension](https://marketplace.visualstudio.com/items?itemName=ms-python.python)
+        1. Open your VSCode settings, in `Code -> Preferences -> Settings`.
+        1. Search for "python formatting provider" and select "black" from the dropdown menu.
```
Comment (jeffreykennethli) [inline onboarding/onboarding.md:86]: nit: link to black docs or repo for those unfamiliar with this formatter

Comment (mvashishtha) [reply onboarding/onboarding.md:86]: Done.

**Comment by YarShev** [comment]
@mvashishtha , please rebase on current master to pass TeamCity runs (failed because of #3648) and also do not forget linking an issue to be resolved in the PR template (I already linked it) for future PRs.

**Comment by mvashishtha** [comment]
> @mvashishtha , please rebase on current master to pass TeamCity runs (failed because of #3648) and also do not forget linking an issue to be resolved in the PR template (I already linked it) for future PRs.

Done. Thanks for linking the issue as well.

**Comment by RehanSD** [comment]
We've discovered that on Mac, conda hangs when creating an environment from the `environment-dev.yml`. This is due to two things:
1. conda's solver is unoptimized
2. installing `ray-default` via conda from the yml file
It is unclear why the second is an issue - after testing we discovered that moving `ray-default` from the conda dependencies to the pip packages greatly speeds up how long conda takes.

To resolve these issues, we have decided to recommend mamba to contributors, as well as moved `ray-default` to the pip install packages section of the yaml file.

File: docs/contributing.rst [hunk 2]
```diff
@@ -13,6 +13,9 @@ want to review in order to get started.
 
 Also, feel free to join the discussions on the `developer mailing list`_.
 
+If you want a quick guide to getting your development enviroment setup, please
+use `the contributing instructions on GitHub`_.
```
Comment (YarShev) [inline docs/contributing.rst:17]: I got 404 page when going by the link. Please, check if it really works.

File: contributing/contributing.md [hunk 3]
```diff
@@ -76,3 +76,16 @@ you should reopen your terminal to find "(base)" next to your prompt: ![](conda_
...
+1. (Optional) We recommend a few workflow settings:
+
+    1. If you use Visual Studio Code, auto-format with [black](https://black.readthedocs.io/en/stable/) every time you save changes:
+        1. Install [Microsoft's Python extension](https://marketplace.visualstudio.com/items?itemName=ms-python.python)
+        1. Open your VSCode settings, in `Code -> Preferences -> Settings`.
+        1. Search for "python formatting provider" and select "black" from the dropdown menu.
+        1. Again in settings, search for "format on save" and enable the "Editor: Format on Save" option.
+    2. Add a pre-commit hook:
+        1. In your modin repository, copy [this pre-commit file](pre-commit) to `.git/hooks/pre-commit`
+        1. Every time you try to commit, git will run flake8 and scripts/doc_checker.py and abort the commit if either fails. This lets you make sure your commits pass these tests before you commit them to GitHub.
```
Comment (YarShev) [inline contributing/contributing.md:90]: I do not see `scripts/doc_checker.py` run in pre-commit file.

**Comment by YarShev** [comment]
> We've discovered that on Mac, conda hangs when creating an environment from the `environment-dev.yml`. This is due to two things:
> 
> 1. conda's solver is unoptimized
> 2. installing `ray-default` via conda from the yml file
>    It is unclear why the second is an issue - after testing we discovered that moving `ray-default` from the conda dependencies to the pip packages greatly speeds up how long conda takes.
> 
> To resolve these issues, we have decided to recommend mamba to contributors, as well as moved `ray-default` to the pip install packages section of the yaml file.

@RehanSD , should we update our `environment-dev.yml` to put `ray-default` to the pip install packages section?

**Comment by RehanSD** [comment]
Hi @YarShev! We did some more testing and discovered that even after moving `ray[default]` to the pip install section conda still hangs sometimes and so we've decided to do some more research, and move that to another PR so as to not keeping blocking this one!

File: contributing/contributing.md (hunk 3 above)
Comment (mvashishtha) [reply contributing/contributing.md]: I deleted the reference to it here.

File: docs/contributing.rst (hunk 2 above)
Comment (mvashishtha) [reply docs/contributing.rst:17]: It should work after this PR, which moves onboarding/onboarding.md to contributing/contributing.md

**Review by RehanSD — APPROVED** [review]
Great work Mahesh! LGTM!

File: docs/contributing.rst (hunk 2 above)
Comment (YarShev) [reply docs/contributing.rst:17]: yep, it works

**Comment by YarShev** [comment]
> Hi @YarShev! We did some more testing and discovered that even after moving `ray[default]` to the pip install section conda still hangs sometimes and so we've decided to do some more research, and move that to another PR so as to not keeping blocking this one!

That makes sense! Could you open an issue to track this?

**Pull request merged**